src/main.rs

```diff
@@ -62,7 +64,9 @@ fn main() {
         let sender = event_loop.channel();
 
         let context = Context::shared(matches.opt_present("v"),
-                                      matches.opt_str("n"));
+                                      matches.opt_str("n"),
+                                      matches.opt_str("p"),
+                                      matches.opt_str("w"),);
```

Comment:
I would rather parse p & w here, to be able to display the usage() if parse<u16> fails. And change Context::new() to take Option<u16>